File: MagicHax/android/app/src/main/java/me/nathanp/magiccreatures/view/ImageRadioButton.java
package me.nathanp.magiccreatures.view;

import android.content.Context;
import android.graphics.drawable.Drawable;
import android.util.AttributeSet;

import androidx.appcompat.widget.AppCompatRadioButton;

public class ImageRadioButton extends AppCompatRadioButton {

    public ImageRadioButton(Context context) {
        super(context);
    }

    public ImageRadioButton(Context context, AttributeSet attrs) {
        super(context, attrs);
    }

    @Override
    protected void onMeasure(int widthMeasureSpec, int heightMeasureSpec) {
        int desiredWidth = MeasureSpec.getSize(widthMeasureSpec);
        Drawable drawable = getBackground();

        float aspectRatio = (float)drawable.getIntrinsicHeight() / (float)drawable.getIntrinsicWidth();
        setMeasuredDimension(desiredWidth, (int)(desiredWidth * aspectRatio));
    }
}

File: MagicHax/android/app/src/main/java/me/nathanp/magiccreatures/view/ImageRadioButton.kt
package me.nathanp.magiccreatures.view

import android.content.Context
import android.util.AttributeSet
import androidx.appcompat.widget.AppCompatRadioButton

class ImageRadioButton : AppCompatRadioButton {
    constructor(context: Context?) : super(context)
    constructor(context: Context?, attrs: AttributeSet?) : super(context, attrs)

    override fun onMeasure(widthMeasureSpec: Int, heightMeasureSpec: Int) {
        val desiredWidth = MeasureSpec.getSize(widthMeasureSpec)
        val desiredHeight = MeasureSpec.getSize(heightMeasureSpec)
        val drawable = background
        val aspectRatio = drawable.intrinsicHeight.toFloat() / drawable.intrinsicWidth.toFloat()
        setMeasuredDimension((desiredHeight * aspectRatio).toInt(), desiredHeight)
    }
}